compiler/rustc_borrowck/src/diagnostics/conflict_errors.rs

```diff
@@ -192,6 +192,146 @@ impl<'cx, 'tcx> MirBorrowckCtxt<'cx, 'tcx> {
                     is_loop_move = true;
                 }
 
+                struct ExpressionFinder<'hir> {
```

Comment:
How self-contained is this new logic? Is it possible to pull it out into a new method?

Reply:
It can :)